dirsrvtests/tests/suites/backups/backup_test.py

```diff
@@ -106,6 +108,65 @@ def test_db_home_dir_online_backup(topo):
         topo.standalone.tasks.db2bak(backup_dir=f'{backup_dir}', args={TASK_WAIT: True})
         assert topo.standalone.ds_error_log.match(f".*Failed renaming {backup_dir}.bak back to {backup_dir}")
 
+def test_replication(topo_m2):
+    """Test that if the dbhome directory is set causing an online backup to fail,
+    the dblayer_backup function should go to error processing section.
+
+    :id: 9c826d36-b17d-11ee-855f-482ae39447e5
+    :setup: Two suppliers
+    :steps:
+        1. Perform backup on S1
+        2. Perform changes on both suppliers
+        3. Wait until replication is in sync
+        4. Stop S1
+        5. Destroy S1 database
+        6. Start S1
+        7. Restore S1 from backup
+        8. Wait until replication is in sync
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+        6. Success
+        7. Success
+        8. Success
+    """
+    S1 = topo_m2.ms["supplier1"]
+    S2 = topo_m2.ms["supplier2"]
+    repl = ReplicationManager(DEFAULT_SUFFIX)
+
+    with tempfile.TemporaryDirectory(dir=S1.ds_paths.backup_dir) as backup_dir:
+        # Step 1: Perform backup on S1
+        # Use the offline method to have a cleanly stopped state in changelog.
+        S1.stop()
+        assert S1.db2bak(backup_dir)
+        S1.start()
+
+        # Step 2:  Perform changes on both suppliers and wait for replication
+        # Note: wait_for_replication perform changes
+        repl.wait_for_replication(S1, S2)
+        repl.wait_for_replication(S2, S1)
+        # Step 4: Stop S1
+        S1.stop()
+        # Step 5: Destroy S1 database
+        if get_default_db_lib() == "mdb":
+            os.remove(f'{S1.ds_paths.db_dir}/data.mdb')
+        else:
+            shutil.rmtree(f'{S1.ds_paths.db_dir}/userRoot')
```

Comment:
A backup should restore all backends. Why only removing userRoot ?
Also the removal of the database is not described in the [doc](https://access.redhat.com/documentation/en-us/red_hat_directory_server/11/html/administration_guide/restoring_directory_server#doc-wrapper) . Is it necessary so the doc should describe it ?

Reply:
In production the removal is not needed: the restore task is supposed to delete the whole db (so the doc is OK), But in this test case, I wanted to be absolutely sure that the db get destroyed to verify that  the "restore" task restores it properly. The test use only a single backend (i.e: userRoot) and in Python it is easier to remove it than to remove everything in the db directory.